plugins/source_metrics.go

```diff
@@ -0,0 +1,104 @@
+package plugins
+
+import (
+	"time"
+
+	"github.com/cloudquery/plugin-sdk/schema"
+)
+
+type SourceMetrics struct {
+	TableClient map[string]map[string]*TableClientMetrics
+}
+
+type TableClientMetrics struct {
+	Resources uint64
+	Errors    uint64
+	Panics    uint64
+	StartTime time.Time
+	EndTime   time.Time
+}
+
+func (s *TableClientMetrics) Equal(other *TableClientMetrics) bool {
```

Comment:
`DeepEqual` is super slow and can have unpredictable results depending on what you are trying to achieve so we want to define equal for every type. DeepEqual is usually used for testing purposes when you don't control or have access to some external struct.

Reply:
My concern is that someone adds a field to the struct and then forgets to add it to the `Equal`.
i.e. We'll have to maintain this implementation. Not sure the performance hit is something that can really slow us down.

Where is the `Equal` function used?

Reply:
we use it in tests but what it the issue of maintaining this and using a library that is used to compare structs which are not in control of the author?

If you look at `SourceMetrics` the `Equal` function is not that simple comparing the `map` but it ensure it will work and I can also add tests to that to ensure we don't forget to update that.

Reply:
If it's only used in the tests I would say we don't need it as a part of the struct and just do the equality in the test (via library or a helper function).

Again this is not blocking for the PR, just seems there's already existing code we can use to compare structs in tests, so we don't need to re-implement it

Reply:
+1 to what @erezrokah said; not a blocker, but I'd vote for go-cmp in tests rather than maintain equality operators if we don't need them in our (non-test) code.